src/convex/ConvexAvatarMultiToken.sol

```diff
@@ -0,0 +1,698 @@
+    EnumerableSetUpgradeable.UintSet internal pidsPrivateVaults;
+
+    /// @dev holds the relation between pid in convex-frax system and private vaults avatar created
+    mapping(uint256 => address) public privateVaults;
+    mapping(address => bytes32[]) public kekIds;
+
+    ////////////////////////////////////////////////////////////////////////////
+    // ERRORS
+    ////////////////////////////////////////////////////////////////////////////
+
+    error NotOwnerOrManager(address caller);
+    error NotKeeper(address caller);
+
+    error InvalidBps(uint256 bps);
+    error LessThanBpsMin(uint256 bpsVal, uint256 bpsMin);
+    error MoreThanBpsVal(uint256 bpsMin, uint256 bpsVal);
+
+    error NothingToDeposit();
+    error NothingToWithdraw();
+    error NoRewards();
+
+    error TooSoon(uint256 currentTime, uint256 updateTime, uint256 minDuration);
+
+    error CurveLpStillStaked(address curveLp, address basePool, uint256 stakingBalance);
+
+    error PoolDeactivated(uint256 pid);
+    error PidNotIncluded(uint256 pid);
+    error NoPrivateVaultForPid(uint256 pid);
+
+    ////////////////////////////////////////////////////////////////////////////
+    // EVENTS
+    ////////////////////////////////////////////////////////////////////////////
+
+    event ManagerUpdated(address indexed newManager, address indexed oldManager);
+    event KeeperUpdated(address indexed newKeeper, address indexed oldKeeper);
+
+    event ClaimFrequencyUpdated(uint256 newClaimFrequency, uint256 oldClaimFrequency);
+
+    event MinOutBpsCrvToWethValUpdated(uint256 newValue, uint256 oldValue);
+    event MinOutBpsCvxToWethValUpdated(uint256 newValue, uint256 oldValue);
+    event MinOutBpsFxsToFraxValUpdated(uint256 newValue, uint256 oldValue);
+    event MinOutBpsWethToUsdcValUpdated(uint256 newValue, uint256 oldValue);
+
+    event Deposit(address indexed token, uint256 amount, uint256 timestamp);
+    event Withdraw(address indexed token, uint256 amount, uint256 timestamp);
+
+    event RewardClaimed(address indexed token, uint256 amount, uint256 timestamp);
+    event RewardsToStable(address indexed token, uint256 amount, uint256 timestamp);
+
+    ////////////////////////////////////////////////////////////////////////////
+    // MODIFIERS
+    ////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Checks whether a call is from the owner or manager.
+    modifier onlyOwnerOrManager() {
+        if (msg.sender != owner() && msg.sender != manager) {
+            revert NotOwnerOrManager(msg.sender);
+        }
+        _;
+    }
+
+    /// @notice Checks whether a call is from the keeper.
+    modifier onlyKeeper() {
+        if (msg.sender != keeper) {
+            revert NotKeeper(msg.sender);
+        }
+        _;
+    }
+
+    function initialize(
+        address _owner,
+        address _manager,
+        address _keeper,
+        uint256[] memory _pids,
+        uint256[] memory _fraxPids
```

Comment:
```suggestion
        uint256[] calldata _pids,
        uint256[] calldata _fraxPids
```

Reply:
resolved in 8f1285e8799cd5cb413b8cb597f1829926afab18